### PR TITLE
printers MAINTENANCE restore json printing to previous style

### PR DESCRIPTION
**Pull request by choppsv1**
Makes 2 changes, to return formatting to what was in libyang1

- Change:
```
    {
      list-node-content
    },    {
      list-node-content
    }
```
 Back to the definitely more common:
```
    {
      list-node-content
    },
    {
      list-node-content
    }
```
- Change leaf lists back to be one item per line when formatting. While
one line might look nice for leaf lists with only a couple items, it
can't be considered "formated" when that list grows to many items
and everything is on a single line wrapping around the screen.

### CODE REVIEW

File: src/printer_json.c [hunk 1]
```diff
@@ -135,12 +135,7 @@ json_print_array_close(struct jsonpr_ctx *ctx)
 
     LEVEL_DEC;
     ly_set_rm_index(&ctx->open, ctx->open.count - 1, NULL);
-    if (schema && (schema->nodetype == LYS_LEAFLIST)) {
```
Comment (rkrejci) [inline src/printer_json.c:138]: The `schema` variable is unused now.

File: src/printer_json.c [hunk 2]
```diff
@@ -98,7 +98,7 @@ static LY_ERR
 json_print_array_open(struct jsonpr_ctx *ctx, const struct lyd_node *node)
 {
     /* leaf-list's content is always printed on a single line */
```
Comment (rkrejci) [inline src/printer_json.c:100]: This comment is no more right.

**Review by rkrejci — CHANGES_REQUESTED** [review]
I see that when doing formatting (indentation, etc.) of the output, it should be done completely including the leaf-list's values. But, besides shrinking everything on a single line with `LYD_PRINT_SHRINK`, does it make sense to keep the current way of formating via some printer option? Let's say `LYD_PRINT_JSON_LEAFLIST_ONELINE`?

**Comment by choppsv1** [comment]
> I see that when doing formatting (indentation, etc.) of the output, it should be done completely including the leaf-list's values. But, besides shrinking everything on a single line with `LYD_PRINT_SHRINK`, does it make sense to keep the current way of formating via some printer option? Let's say `LYD_PRINT_JSON_LEAFLIST_ONELINE`?

I don't think it's worth it. These types of choices typically are made based on tty columns etc, seems not worth the complexity in the API. Perhaps you could open an issue tracking that enhancement idea since adding flags would be backward compatible at some future date.

File: src/printer_json.c (hunk 2 above)
Comment (choppsv1) [reply src/printer_json.c]: fixed

File: src/printer_json.c (hunk 1 above)
Comment (choppsv1) [reply src/printer_json.c:138]: fixed

**Comment by choppsv1** [comment]
@rkrejci the CI failure seems to be related to the network, not the change. Can you please rerun the workflow?

**Comment by rkrejci** [comment]
> Perhaps you could open an issue tracking that enhancement idea since adding flags would be backward compatible at some future date.

That's fair, I'm going to do that.

**Review by rkrejci — APPROVED** [review]

**Pull request merged**